flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/expressions/converter/CustomizedConvertRule.java

```diff
@@ -0,0 +1,361 @@
+			DataType type = ((TypeLiteralExpression) children.get(1)).getOutputDataType();
+			SqlThrowExceptionFunction function = new SqlThrowExceptionFunction(
+					context.getTypeFactory().createFieldTypeFromLogicalType(fromDataTypeToLogicalType(type)));
+			return context.getRelBuilder().call(function, context.toRexNode(children.get(0)));
+		});
+	}
+
+	@Override
+	public Optional<RexNode> convert(CallExpression call, ConvertContext context) {
+		Conversion conversion = DEFINITION_RULE_MAP.get(call.getFunctionDefinition());
+		if (conversion == null) {
+			return Optional.empty();
+		} else {
+			return Optional.of(conversion.convert(call.getChildren(), context));
+		}
+	}
+
+	private static RexNode convertCast(List<Expression> children, ConvertContext context) {
+		RexNode child = context.toRexNode(children.get(0));
+		TypeLiteralExpression type = (TypeLiteralExpression) children.get(1);
+		return context.getRelBuilder().getRexBuilder().makeAbstractCast(
+			context.getTypeFactory().createFieldTypeFromLogicalType(
+				type.getOutputDataType().getLogicalType().copy(child.getType().isNullable())),
+			child);
+	}
+
+	private static RexNode convertArrayElement(List<Expression> children, ConvertContext context) {
+		List<RexNode> childrenRexNode = context.toRexNodes(children);
+		return context.getRelBuilder().call(FlinkSqlOperatorTable.ELEMENT, childrenRexNode);
+	}
+
+	private static RexNode convertOrderAsc(List<Expression> children, ConvertContext context) {
+		List<RexNode> childrenRexNode = context.toRexNodes(children);
+		return childrenRexNode.get(0);
+	}
+
+	private static RexNode convertTimestampDiff(List<Expression> children, ConvertContext context) {
+		List<RexNode> childrenRexNode = context.toRexNodes(children);
+		return context.getRelBuilder().call(FlinkSqlOperatorTable.TIMESTAMP_DIFF, childrenRexNode.get(0), childrenRexNode.get(2),
+			childrenRexNode.get(1));
+	}
+
+	private static RexNode convertIsNull(List<Expression> children, ConvertContext context) {
+		List<RexNode> childrenRexNode = context.toRexNodes(children);
+		return context.getRelBuilder().isNull(childrenRexNode.get(0));
+	}
+
+	private static RexNode convertNotBetween(List<Expression> children, ConvertContext context) {
+		List<RexNode> childrenRexNode = context.toRexNodes(children);
+		Preconditions.checkArgument(childrenRexNode.size() == 3);
+		RexNode expr = childrenRexNode.get(0);
+		RexNode lowerBound = childrenRexNode.get(1);
+		RexNode upperBound = childrenRexNode.get(2);
+		return context.getRelBuilder().or(
+			context.getRelBuilder().call(FlinkSqlOperatorTable.LESS_THAN, expr, lowerBound),
+			context.getRelBuilder().call(FlinkSqlOperatorTable.GREATER_THAN, expr, upperBound));
+	}
+
+	private static RexNode convertBetween(List<Expression> children, ConvertContext context) {
+		List<RexNode> childrenRexNode = context.toRexNodes(children);
+		Preconditions.checkArgument(childrenRexNode.size() == 3);
```

Comment:
How about we use `TableException` here? Also there are some Conversions that do not check the number of arguments. We should do that in all Conversions.